homeassistant/components/camera/__init__.py

```diff
@@ -154,6 +155,57 @@ class Image:
     content: bytes = attr.ib()
 
 
+@dataclass
+class RtcConfiguration:
+    """Web Rtc settings for peer connections.
+
+    Subset of the full RtcConfiguration interface definition
+    https://devdoc.net/web/developer.mozilla.org/en-US/docs/Web/API/RTCConfiguration.html
+    """
+
+    @dataclass
+    class IceServer:
+        urls: str
+        credential: str | None = None
+        username: str | None = None
+
+    class BundlePolicy(StrEnum):
+        BALANCED = auto()
+        MAX_BUNDLE = "max-bundle"
+        MAX_COMPAT = "max-compat"
+
+    class IceTransportPolicy(StrEnum):
+        ALL = auto()
+        RELAY = auto()
+
+    ice_servers: list[IceServer] | None = None
+    ice_transport_policy: IceTransportPolicy | None = None
+    bundle_policy: BundlePolicy | None = None
+
+    def as_ws_dict(self) -> dict:
+        """Return a dictionary conforming to the RtcConfiguration interface with null values omitted."""
+        return asdict(
+            self,
+            dict_factory=lambda x: {
+                camelcase(k): str(v) if isinstance(v, StrEnum) else v
+                for (k, v) in x
+                if v is not None
+            },
+        )
```

Comment:
We call `dataclasses.asdict` on ourselves, why don't we do that on the `ice_servers` list? How come we don't omit null values from `IceServer` items?

Also, Instead of the custom `dict_factory`, would it be more maintainable and easier to understand if the three items we intend to serialize were just spelled out?